Twitter Communication methods

DanielEizenberger opened the pull request.

Germiee committed:
Added initial python files

DanielEizenberger committed:
Implemented command line arguments

Germiee committed:
Added __init__ method

Germiee committed:
Merge branch 'Development' of github.com:JulianSchwaiger/RandomTwitterBotV1 into Development

DanielEizenberger committed:
Removed seconds for the tweet interval

DanielEizenberger committed:
Merge branch 'master' into Development

DanielEizenberger committed:
Called the twitter communications init function

DanielEizenberger committed:
Added the loguru dependency

Germiee committed:
Potential fix for twitter connection

:100:

Germiee committed:
Swapped from python-twitter to tweepy

:fire:

DanielEizenberger committed:
Added a connection to firebase

DanielEizenberger committed:
Added basic communication with firestore

Added functions to:
- Fetch a tweet
- Fetch a tweet and set it's "HasBeenTweeted" to true
- Add a tweet to the database

DanielEizenberger committed:
Added functions to count tweets

Added functions to:
- count all "untweeted" tweets
- count all tweets
- count all "untweeted" tweets by a specific user
- count all tweets by a specific user

Germiee reviewed (approved):
Fetch tweet could possibly be randomized so that the tweets don't get tweeted in a specific order, but for now this is fine. 💯 🔥 👍 🥇

Germiee committed:
added getMessage implementation

DanielEizenberger committed:
Updated Dependencies

DanielEizenberger committed:
Merge branch 'Development' of https://github.com/JulianSchwaiger/RandomTwitterBotV1 into Development

Germiee committed:
added better error handling to twitter_communications.py

Germiee committed:
Merge branch 'Development' of github.com:JulianSchwaiger/RandomTwitterBotV1 into Development

Germiee committed:
More error handling changes

DanielEizenberger committed:
Updated function names

DanielEizenberger committed:
Merge branch 'Development' of https://github.com/JulianSchwaiger/RandomTwitterBotV1 into Development

Germiee committed:
more logging

better logging for errors and connection to api

Germiee committed:
Changed tweepy auth method

Now using OAuth 2.0  with the Bearer Token

Germiee committed:
try making a tweet using new api

Germiee committed:
parameter now specified, changed python version

Germiee committed:
python version changing

Germiee committed:
changed authentication method again

now using OAuth 1.0a

Germiee committed:
more authentication changes again

Twitter API V2 does not have proper requests for direct messages yet so we have to go back to V1.1

Germiee committed:
api version also means other method again.

Germiee committed:
global variable fix

Germiee committed:
getting direct messages updated

getting forbidden error from the api, trying to find the cause

Germiee committed:
init method now returns api connection

Germiee committed:
the api blocks duplicate tweets

Germiee committed:
removed cursor for pagination

Germiee committed:
wrong data structure for direct messages

Germiee committed:
added return to get_new_messages

Germiee committed:
Error handling, added delete_message

Germiee committed:
fixed type, added send message method

Germiee committed:
twitter does numbers as strings.

DanielEizenberger commented on src/twitter_communications.py:
```diff
@@ -0,0 +1,82 @@
+
+        new_messages = []
+        for mes in all_messages:
+            if mes.message_create["sender_id"] != "1332712271987023874" :
```
Please replace the string ("1332712271987023874") with a global variable

DanielEizenberger commented on src/twitter_communications.py:
```diff
@@ -0,0 +1,82 @@
+        if hasattr(e, 'message'):
+            logger.error("Error message: {}", e.message)
+        else:
+            logger.error("Error message: {}", e) 
```
New line is missing

Germiee committed:
added default_bot_id variable

Germiee committed:
added get_user_name method

Germiee committed:
🙄🙄

Germiee committed:
why is twitters api so weird :(

DanielEizenberger commented on src/twitter_communications.py:
```diff
@@ -85,4 +85,17 @@ def send_message(client, id, content):
+        if hasattr(e, 'message'):
+            logger.error("Error message: {}", e.message)
+        else:
+            logger.error("Error message: {}", e) 
```
New line is missing

Germiee committed:
Added do_tweet method which does the tweeting

Germiee committed:
Finished twitter_communications.py main methods and logic

JulianSchwaiger commented:
We should squash this, 45 commits with often many small testing based changes are not good to have on our main branch and only cause utter confusion

Main reason for this was the API keys etc so this shouldn't be an issue anymore, atleast not to this extent
